akka-projection-core/src/main/scala/akka/projection/scaladsl/OffsetStore.scala

```diff
@@ -4,22 +4,34 @@
 
 package akka.projection.scaladsl
 
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
 import akka.Done
 
-import scala.concurrent.{ExecutionContext, Future}
-import scala.language.higherKinds
+object OffsetStore {
+  sealed trait Strategy
+  case object NoOffsetStorage extends Strategy
+  case object AtMostOnce extends Strategy
+  final case class AtLeastOnce(afterNumberOfEvents: Int, orAfterDuration: FiniteDuration) extends Strategy
+  case object OnceAnOnlyOnce extends Strategy
 
-trait OffsetStore[Offset, IO] {
-  def readOffset(): Future[Option[Offset]]
-  def saveOffset(offset: Offset): IO
-}
+  private val _noOffsetStore = new OffsetStore[Any] {
+    override def readOffset(): Future[Option[Any]] = Future.successful(None)
+    override def saveOffset(offset: Any): Future[Done] = Future.successful(Done)
```

Comment:
I find it strange it returns a Future. It forces it to run on a different thread.

Reply:
You are right that there are too many futures here. Not needed for things that are specific to JDBC. Changed in 00efbc7